### PR TITLE
proxy: only divert outbound traffic from Ingress proxy to cilium_host

### PR DESCRIPTION
**Pull request by julianwiedmann**
https://github.com/cilium/cilium/pull/29530 re-introduced the custom routing of Ingress proxy traffic to `cilium_host`. But we actually don't need to route *all* Ingress proxy traffic - it's sufficient to divert traffic that is in direction from the connection's source to destination.

### CODE REVIEW

**Review by jschwinger233 — APPROVED** [review]

**Review by jschwinger233 — CHANGES_REQUESTED** [review]
Hang on a moment, let me confirm some details.

**Review by jschwinger233 — APPROVED** [review]
No concerns for ingress, thanks!

File: pkg/proxy/routes.go [hunk 1]
```diff
@@ -99,8 +99,8 @@ var (
 	// Routing rule for traffic from proxy.
 	fromProxyRule = route.Rule{
 		Priority: linux_defaults.RulePriorityFromProxyIngress,
-		Mark:     linux_defaults.MagicMarkIsProxy,
-		Mask:     linux_defaults.MagicMarkHostMask,
+		Mark:     linux_defaults.MagicMarkIngress,
+		Mask:     linux_defaults.MagicMarkProxyNoIDMask,
```
Comment (jschwinger233) [inline pkg/proxy/routes.go]: MagicMarkProxyNoIDMask int = 0xFFFFFEFF, this patch is also trying to route egress packets while original one touches ingress only. Maybe make it clear in PR title and commit message?

Comment (julianwiedmann) [reply pkg/proxy/routes.go]: yes, that's a bug I think. Let me fix that up quickly ...

**Review by jschwinger233 — APPROVED** [review]
:+1:

File: pkg/proxy/routes.go (hunk 1 above)
Comment (julianwiedmann) [reply pkg/proxy/routes.go]: thank you Gray!

**Pull request closed without merging**